src/main/java/lesson8Person/Person.java

```diff
@@ -0,0 +1,71 @@
+package lesson8Person;
+
+import java.time.*;
+import java.util.Objects;
+
+public class Person {
+    private final String name;
+    private final String surname;
+
+    // if I don't initialize these, I can't use the 'LocalDate.of' method, any solutions to that?
+    private int yearOfBirth = 0;
+    private Month monthOfBirth = Month.January;
+    private int dayOfBirth = 0;
+
+    private final LocalDate birthDate;
```

Comment:
To pole duplikuje logikę pól: yearOfBirth, monthOfBirth oraz dayOfBirth. Nie chcemy przechowywać tych samych danych w kilku miejscach.

Reply:
To jak użytkownik wprowadzi swoje dane, a jak utworzysz obiekt Person to dwa różne tematy. Nic nie stoi na przeszkodzie, aby pobrać te dane od użytkownika osobno, a następnie przy tworzeniu obiektu klasy Person wykorzystać LocalDate.of przed wywołaniem konstruktora.